src/Graphex.hs

```diff
@@ -53,6 +55,9 @@ graphToDep (Graph m attrs) = GraphDef {
 getInput :: FilePath -> IO (Graph Text)
 getInput fn = either fail (pure . depToGraph) . eitherDecode =<< BL.readFile fn
 
+hGetInput :: Handle -> IO (Graph Text)
+hGetInput h = either fail (pure . depToGraph) . eitherDecode =<< BL.hGetContents h
+
```

Comment:
I could've abstracted more here (files can be Handles too), but it didn't feel worth it.